### PR TITLE
website/integrations: Add Semgrep

### PR DESCRIPTION
**Pull request by Infernogeek1**
## Details
Added Semgrep integration. Some things said there I found out a hard way.
1. EC certs did not quite work out
2. it requires SAML property mappings
---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

### CODE REVIEW

**Review by dominic-r — CHANGES_REQUESTED** [review]

File: website/integrations/services/semgrep/index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+5. Navigate to **System** -> **Certificates**.
+6. Generate a new RSA certificate, pick a name.
+7. Download the generated certificate, as you will need it later.
+8. Create a new SAML provider under **Applications** -> **Providers** using the following settings:
+    - **ACS URL**: `https://semgrep.dev/api/auth/saml/devcompany/`
+    - **Issuer**: `https://authentik.company`
+    - **Audience**: `semgrep-dev`
+    - **Service Provider Binding**: `Post`
+    - **Signing Keypair**: Select the RSA certificate you have generated previously.
+    - **Property mappings**: Select `semgrep-name` and `semgrep-email` mappings, replacing authentik-provided email and name mappings. Keep other mappings default.
```
[the remark this answers is not in the document]
Comment (Infernogeek1) [reply website/integrations/services/semgrep/index.md:40]: I will remove that. The SAML login worked fine with these two mappings.

File: website/integrations/services/semgrep/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+## Preparation
+
+The following placeholders will be used:
+
+-   `authentik.company` is the FQDN of the authentik install.
+-   `devcompany` is the organization name on Semgrep Cloud platform.
+
+## authentik configuration
+
+1. Login into authentik instance as an administrator.
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:22]:
```suggestion
1. Log in to your authentik instance as an administrator.
```

File: website/integrations/services/semgrep/index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+7. Generate a new RSA certificate.
+8. Download the generated certificate, as you will need it later.
+9. Create a new SAML provider under **Applications** -> **Providers** using the following settings:
+    - **ACS URL**: `https://semgrep.dev/api/auth/saml/devcompany/`
+    - **Issuer**: `https://authentik.company`
+    - **Audience**: `semgrep-dev`
+    - **Service Provider Binding**: `Post`
+    - **Signing Keypair**: Choose the RSA certificate you generated earlier.
+    - **Property mappings**: `semgrep-name` and `semgrep-email`
+10. Create a new application under **Applications** -> **Applications**, pick a name and a slug, and assign the provider that you have just created.
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:43]:
```suggestion
10. Create a new application under **Applications** -> **Applications**, pick a name and a slug, and assign the provider that you just created.
```

File: website/integrations/services/semgrep/index.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Semgrep Configuration
+
+1. Login to Semgrep Cloud platform as an administrator.
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:47]:
```suggestion
1. Log in to Semgrep Cloud platform as an administrator.
```

File: website/integrations/services/semgrep/index.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    - **IdP Issuer ID**: `https://authentik.company`
+    - **Upload/paste certificate**: Downloaded from the previous step.
+
+## Verification
+
+1. Open an Incaognito window and navigate to `https://semgrep.dev/login`
+2. Click **Use SSO** on the login screen.
+3. Enter the email address associated with the domain you provided earlier.
+4. Login with authentik.
+5. You will be redirected to home screen of Semgrep Cloud platform.
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:64]:
```suggestion
5. You will be redirected to the home screen of Semgrep Cloud platform.
```

File: website/integrations/services/semgrep/index.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    - **IdP SSO URL**: `https://authentik.company/application/saml/<semgrep slug>/sso/binding/post/`
+    - **IdP Issuer ID**: `https://authentik.company`
+    - **Upload/paste certificate**: Downloaded from the previous step.
+
+## Verification
+
+1. Open an Incaognito window and navigate to `https://semgrep.dev/login`
+2. Click **Use SSO** on the login screen.
+3. Enter the email address associated with the domain you provided earlier.
+4. Login with authentik.
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:63]:
```suggestion
4. Log in with authentik.
```

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks @Inferno-geek for another integration! A couple of nits; "log in" versus "login" is a beast... checkout our new-ish [Style Guide](https://docs.goauthentik.io/developer-docs/docs/style-guide) for tips on when to use which.

**Comment by Infernogeek1** [comment]
I had a slight misconfiguration of my git. Oops.

File: website/integrations/services/semgrep/index.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+9. Create a new SAML provider under **Applications** -> **Providers** using the following settings:
+    - **ACS URL**: `https://semgrep.dev/api/auth/saml/devcompany/`
+    - **Issuer**: `https://authentik.company`
+    - **Audience**: `semgrep-dev`
+    - **Service Provider Binding**: `Post`
+    - **Signing Keypair**: Choose the RSA certificate you generated earlier.
+    - **Property mappings**: `semgrep-name` and `semgrep-email`
+10. Create a new application under **Applications** -> **Applications**, pick a name and a slug, and assign the provider that you just created.
+
+## Semgrep Configuration
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:45]:
```suggestion
## Semgrep configuration
```

Comment (tanberry) [reply website/integrations/services/semgrep/index.md:45]: Ooops, sorry I missed this one earlier. Can you add it real quick please @Infernogeek1 , then I'll merge.

File: website/integrations/services/semgrep/index.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+5. Fill in the following:
+    - **Display name**: Anything you like.
+    - **Email domain**: `company`
+    - **IdP SSO URL**: `https://authentik.company/application/saml/<semgrep slug>/sso/binding/post/`
+    - **IdP Issuer ID**: `https://authentik.company`
+    - **Upload/paste certificate**: Downloaded from the previous step.
+
+## Verification
+
+1. Open an Incaognito window and navigate to `https://semgrep.dev/login`
```
Comment (tanberry) [inline website/integrations/services/semgrep/index.md:60]:
```suggestion
1. Open an Incognito window and navigate to `https://semgrep.dev/login`
```

**Review by tanberry — CHANGES_REQUESTED** [review]
Two things I missed in earlier review... please and thanks!

**Comment by Infernogeek1** [comment]
I don't exactly get what is meant by https://github.com/goauthentik/authentik/pull/10849#discussion_r1714167605.; Mind clarifying, please? @tanberry

**Review by tanberry — APPROVED** [review]
Thank you!

**Comment by tanberry** [comment]
> I don't exactly get what is meant by [#10849 (comment)](https://github.com/goauthentik/authentik/pull/10849#discussion_r1714167605).; Mind clarifying, please? @tanberry

I meant that I had missed the capitalization of "configuration", which needed to be lower-case. All looks great now, thanks you, I'll merge!

BTW I hope it is it helpful that I use the Suggestion feature on my edits, so that you can simply accept (or deny haha) the edits.

**Comment by Infernogeek1** [comment]
I mostly edit in Vim, but yes, seeing the diffs helps.

**Pull request merged**